gtsfm/scene_optimizer.py

```diff
@@ -362,55 +364,54 @@ def get_gtsfm_data_with_gt_cameras_and_est_tracks(
+        images: Input images.
+        ba_input_data: GtsfmData input to bundle adjustment.
+        ba_output_data: GtsfmData output to bundle adjustment.
         results_path: Path to directory where GTSFM results will be saved.
-
-    Returns:
-        A list of delayed objects after saving the input and outputs to bundle adjustment.
     """
-    saving_graph_list = []
-    # Save a duplicate in REACT_RESULTS_PATH.
-    for output_dir in [results_path, REACT_RESULTS_PATH]:
-        # Save the input to Bundle Adjustment (from data association).
-        saving_graph_list.append(
-            dask.delayed(io_utils.export_model_as_colmap_text)(
-                gtsfm_data=ba_input_graph,
-                images=image_graph,
-                save_dir=os.path.join(output_dir, "ba_input"),
-            )
-        )
-        # Save the output of Bundle Adjustment.
-        saving_graph_list.append(
-            dask.delayed(io_utils.export_model_as_colmap_text)(
-                gtsfm_data=ba_output_graph,
-                images=image_graph,
-                save_dir=os.path.join(output_dir, "ba_output"),
-            )
-        )
+    start_time = time.time()
+
+    output_dir = results_path
+    # Save the input to Bundle Adjustment (from data association).
+    io_utils.export_model_as_colmap_text(
+        gtsfm_data=ba_input_data,
+        images=images,
+        save_dir=os.path.join(output_dir, "ba_input"),
+    )
 
-        # Save the ground truth in the same format, for visualization.
-        # We use the estimated tracks here, with ground truth camera poses.
-        gt_gtsfm_data = dask.delayed(get_gtsfm_data_with_gt_cameras_and_est_tracks)(cameras_gt, ba_output_graph)
-        saving_graph_list.append(
-            dask.delayed(io_utils.export_model_as_colmap_text)(
-                gtsfm_data=gt_gtsfm_data,
-                images=image_graph,
-                save_dir=os.path.join(output_dir, "ba_output_gt"),
-            )
-        )
+    # Save the output of Bundle Adjustment.
+    io_utils.export_model_as_colmap_text(
+        gtsfm_data=ba_output_data,
+        images=images,
+        save_dir=os.path.join(output_dir, "ba_output"),
+    )
+
+    # Save the ground truth in the same format, for visualization.
+    # We use the estimated tracks here, with ground truth camera poses.
+    gt_gtsfm_data = get_gtsfm_data_with_gt_cameras_and_est_tracks(cameras_gt, ba_output_data)
+
+    io_utils.export_model_as_colmap_text(
+        gtsfm_data=gt_gtsfm_data,
+        images=images,
+        save_dir=os.path.join(output_dir, "ba_output_gt"),
+    )
 
-    return saving_graph_list
+    end_time = time.time()
+    duration_sec = end_time - start_time
+    logger.info("GtsfmData I/O took %.2f sec.", duration_sec)
+    # Delete old version of React results directory.
+    shutil.rmtree(REACT_RESULTS_PATH)
+    # Save a duplicate copy of the directory in REACT_RESULTS_PATH.
+    shutil.copytree(src=results_path, dst=REACT_RESULTS_PATH)
```

Comment:
could you include this copying in the I/O time above as well?

Reply:
Good catch, updated now to include this React I/O time.